lib/utils/location.js

```diff
@@ -75,18 +76,24 @@ const appPageTypes: { [AppType]: {| default?: PageType, pageTypes: PageType[] |}
 		pageTypes: ['wiki', 'search', 'stylesheet', 'modqueue', 'subredditAbout', 'comments', 'commentsLinklist', 'profile', 'liveThread', 'inbox', 'submit', 'account', 'prefs'],
 	},
 	d2x: {
-		pageTypes: ['profile2x', 'profileCommentsPage'],
+		pageTypes: ['profile2x', 'profileCommentsPage', 'submit'],
 	},
 	options: {
 		pageTypes: [],
 	},
 };
 
-export const pageType = _.once((): ?PageType => {
+export const pageType = _.memoize((): ?PageType => {
+	clearPageTypeCache();
 	const spec = appPageTypes[appType()];
 	return spec.pageTypes.find(pageType => regexes[pageType].test(location.pathname)) || spec.default;
 });
 
+function clearPageTypeCache() {
+	waitForEvent(document, 'reddit.urlChanged').then(clearPageTypeCache);
```

Comment:
Nit: It would be cleaner to reset the cache inlined in the function above, like in the example I provided above.